docs: add mustache example for conditional prompt (#830)
langchain-ai/langsmith-docs: docs/prompt_engineering/concepts/index.mdx

Hello, {{name}}!
```

For conditional mustache prompt:

```python
{{#is_logged_in}}
  Welcome back, {{name}}!
{{else}}
  Please log in.
{{/is_logged_in}}
```

- For this case, the playground UI will pick up `is_logged_in` variable,
  but you need to manually specify your vairables in inputs, for example,
  paste this into inputs to make above conditional prompt works

```
{
  "name": "Alice"
}
```

:::tip Mustache format

> LangSmith Playground uses f-string as default Template format,
> but you can switch to Mustache format through Prompt Setting - Template Format setting.
> It gives your more flexbility around conditional variables, loops, and nested keys.
> For conditional variables, you might need to manually edit json variables in 'inputs' section.

Reviewer: ```suggestion
The LangSmith Playground uses `f-string` as the default template format,
but you can switch to `mustache` format in the prompt settings/template format section.
`mustache` gives you more flexibility around conditional variables, loops, and nested keys.
For conditional variables, you'll need to manually add json variables in the 'inputs' section.
```
